### ISSUE TITLE
ngAnimate introduces delays in ngView changes

### ISSUE THREAD

**Issue by drcapulet** [angular/angular.js#14678]
**Do you want to request a *feature* or report a *bug*?**

This is a bug that occurs when simply depending on the `ngAnimate` model, even if you disable it via `$animate.enabled(false)` or `$animateProvider.classNameFilter(...)`

**What is the current behavior?**

https://plnkr.co/edit/yPTvrirAhTMKUPp8F8rz?p=preview

When clicking between the types an alert will appear and you'll see that both `<ng-views>` are visible.

**What is the expected behavior?**

https://plnkr.co/edit/8wQ0dayOXq64ALDDwTTf?p=preview

When performing the same actions, notice how the ng-view is swapped out instantaneously. The only difference in the two examples is this:

```
angular.module('ngViewExample', ['ngRoute', 'ngAnimate'])
```
vs

```
angular.module('ngViewExample', ['ngRoute'])
```

**What is the motivation / use case for changing the behavior?**

It's unexpected behavior that's impossible to disable simply by relying on `ngAnimate` - for example if you want animations when using https://github.com/angular-ui/bootstrap.

**Which versions of Angular, and which browser / OS are affected by this issue? Did this work in previous versions of Angular? Please also test with the latest stable and snapshot (https://code.angularjs.org/snapshot/) versions.**

Tested in 1.3.20, 1.4.9 and 1.5.5 in Chrome.

**Other information (e.g. stacktraces, related issues, suggestions how to fix)**

I have [this commit](https://github.com/angular/angular.js/commit/759ee2285708e37017726be085070574aab96f6d) which flips the ordering of execution of `$animate.enter` and `$animate.leave` which solves our specific issue (we make rendering decisions based on if a given directive is visible by the user) but I fear might cause unintended side effects as it re-orders the divs.

**Comment by gkalpak** [comment]
This is expacted behavior. `$animate` introduces some asychronicity and there is no guarantee that the previous view will be removed or not when the new view's controller is instantiated.

That said, you _can_ disable it, e.g. using `$animate.classNameFilter()`: [Updated plnkr 1][1]
Alternatively, you can wrap your call to `alert` inside a `$timeout` to make sure it called after the DOM operations are over (assuming there is no actual animation): [Updated plnkr 2][2]

--
Closing since this working as expected.

[1]: https://plnkr.co/edit/iSGNfK99fMqy4gqlp0zT?p=preview
[2]: https://plnkr.co/edit/kfNPMmEUqOPPQExcVYZp?p=preview

**Comment by drcapulet** [comment]
@gkalpak Thanks for the quick response - I had only tested `$animate.classNameFilter()` in 1.3 (ex: https://plnkr.co/edit/u3F0IIHCbuCWV5sPx4Ow?p=preview on 1.3.20) - it looks like it was fixed in 1.4.0.

Sorry for the noise!
